internal/testsuite/app.go

```diff
@@ -188,7 +188,10 @@ func (a *App) RunTests(ctx context.Context, testSpecs []*api.TestSpec) (*TestSui
 			eventLogger:          eventLogger,
 		}
 		go func() {
-			testRunner.Run(ctx)
+			err := testRunner.Run(ctx)
+			if err != nil {
+				fmt.Fprintf(a.Out, "error in test case %s: %s", rv.TestCaseReports[i].TestSpec.Name, err)
```

Comment:
This will result in a nil reference panic.
You need to get the test case name from the `testSpecs` passed into the `RunTests` function.

Reply:
Good call.